pkg/resources/masking_policy.go

```diff
@@ -2,282 +2,339 @@ package resources
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
 
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/schemas"
 
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 var maskingPolicySchema = map[string]*schema.Schema{
-	"or_replace": {
-		Type:                  schema.TypeBool,
-		Optional:              true,
-		Default:               false,
-		Description:           "Whether to override a previous masking policy with the same name.",
-		DiffSuppressOnRefresh: true,
-		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-			return old != new
-		},
-	},
-	"if_not_exists": {
-		Type:                  schema.TypeBool,
-		Optional:              true,
-		Default:               false,
-		Description:           "Prevent overwriting a previous masking policy with the same name.",
-		DiffSuppressOnRefresh: true,
-		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-			return old != new
-		},
-	},
 	"name": {
-		Type:        schema.TypeString,
-		Required:    true,
-		Description: "Specifies the identifier for the masking policy; must be unique for the database and schema in which the masking policy is created.",
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("Specifies the identifier for the masking policy; must be unique for the database and schema in which the masking policy is created."),
```

Comment:
blocklisted characters description could be also used on database and schema

Reply:
Done.